cashu/wallet/api/router.py

```diff
@@ -123,6 +123,50 @@ async def invoice(
     return
 
 
+@router.post("/swap", name="Multi-mint swaps", summary="Swap funds between mints.")
+async def swap(
+    amount: int = Query(default=..., description="Amount to swap between mints"),
+    outgoing_mint: str = Query(default=..., description="URL of outgoing mint"),
+    incoming_mint: str = Query(default=..., description="URL of incoming mint"),
+):
+    assert settings.lightning, HTTPException(
+        status_code=status.HTTP_400_BAD_REQUEST,
+        detail="lightning not enabled",
+    )
+    assert outgoing_mint not in incoming_mint, HTTPException(
+        status_code=status.HTTP_400_BAD_REQUEST,
+        detail="mints for swap have to be different",
+    )
```

Comment:
Using `assert` will crash the server completely, no? Instead these should be `raise`:

```suggestion
    if not settings.lightning:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail="lightning not enabled",
        )
    if not outgoing_mint not in incoming_mint:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail="mints for swap have to be different",
        )
```

Would also suggest `"This mint does not support Lightning"` instead of `"lightning not enabled"` as it's more descriptive for a user?

(Also could use **501 Not Implemented** or **403 Forbidden** for the first check ?)

Reply:
Unless `AssertionError` gets caught somewhere but almost always that's a no-no

Reply:
I actually thought you can't raise custom exceptions with an `assert`. However, it will not crash the server, just the thread on which the endpoint lives.